### DISCUSSION TITLE
Animator NPC : is is possible to change military characteristics of NPCs ? and using LLM?

### DISCUSSION THREAD

**Discussion by vramULB** [cmu-sei/GHOSTS#347] [Q&A] [answered]
Hello, I'd like to use the animator to create personas linked to an IT company where I've defined the orginogram. Is this possible?
Using the API, I've seen that the rank (MilitaryBranch enumation) is the only blocking point for the moment, but I'm wondering if this could cause problems with the other components. 

If I create multiple defaults NPCs with the animator using the military characteristics, is it possible to change these characteristics with the LLM?

**Comment by sei-dupdyke** [comment, marked as answer]
Hi, thanks for checking out the project. You should be able to create NPCs with some branch affiliation, but then ignore those properties. There are no hard dependencies on these within the system.

I can remove having to have MilitaryBranch at all at some point in the future.

**Reply by vramULB** [reply]
Thank you for the fast answer, actually I use the /api/npcsgenerate
```
{
  "campaign": "Team 2024",
  "enclaves": [
    {
      "name": "Whatever ICT",
      "teams": [
        {
	  "name": "Top management",
          "machineNameTemplate": "eng{machine_number}",
          "domainTemplate": "eng{machine_number}-janus-ict.be",
          "preferenceSettings": null,
          "npcs": {
            "number": 3,
            "configuration": {
              "branch": "USARMY",
              "rankDistribution": [],
              "unit": "Top management",
              "username": null,
              "preferenceSettings": null
            }
          }
        },
        {
	  "name": "Administration",
          "machineNameTemplate": "eng{machine_number}",
          "domainTemplate": "eng{machine_number}-janus-ict.be",
          "preferenceSettings": null,
          "npcs": {
            "number": 2,
            "configuration": {
              "branch": "USARMY",
              "rankDistribution": [],
              "unit": "Administration",
              "username": "testuser",
              "preferenceSettings": null
            }
          }
        },
      ]
    }
  ]
}
```

There is a way to link the NPCs created to machines ? 

Edit: Using the previous json, I notice that the username, domainTemplate and machineNameTemplate don't work correctly.

**Comment by sei-dupdyke** [comment]
Created issue #351 to address this. Thank you for researching.
